app/src/main/java/com/example/movieapp/data/MovieRepository.kt

```diff
@@ -0,0 +1,28 @@
+package com.example.movieapp.data
+
+import android.util.Log
+import com.example.movieapp.models.MovieDto
+
+class MovieRepository (
+    private val localDataSource: MovieLocalDataSource,
+    private val remoteDataSource: MovieRemoteDataSource
+) {
+
+    suspend fun loadPopularMovies(page: Int = 0): List<MovieDto> {
+        return localDataSource.getPopular(page) ?: run {
+            val response = remoteDataSource.getsPopular(page = page, apiKey = "76422ff67ea5c659115400f1f574c025")
```

Comment:
Extract api key to a constant or enviroment variable.

see comment above.

Reply:
https://stackoverflow.com/questions/35722904/saving-the-api-key-in-gradle-properties

If you don't succeed at this just put it in a `Constants.kt` file, and add it to .gitignore

Reply:
Done